### PR TITLE
Increase the diagonal wire version

### PR DESCRIPTION
**Pull request by myliu-hub**
Inclined wire version added：

1. The location of the XML file for the diagonal wire version of the separate drift chamber is Detector/DetDriftChamber/compact/det_inclined_wire.xml ;
2. The location of the XML file is Detector/DetCRD/compact/CRD_common_v01/DC_Simple_v01_03.xml ;
3. Alpha is 12 degrees ;

### CODE REVIEW

**Comment by myliu-hub** [comment]
@fucd Hi, I have changed it according to your request, do you think it is okay?

**Comment by mirguest** [comment]
Hi @myliu-hub , 
I have a quick look at your PR. It looks like you duplicate the detector constructor c++ code. It will cause problems, especially when you need to synchronize parameters or implementations between the two constructors.
Actually the only difference is how to construct the layer and wire. I would suggest you to only maintain one c++ constrcutor and you can use some software design to reduce such problem. 
Tao

**Comment by fucd** [comment]
@myliu-hub Thank you for update. I have no more comments now. 
> @fucd Hi, I have changed it according to your request, do you think it is okay?

**Comment by wenxingfang** [comment]
@ihepzhangyao @mirguest  This PR seems quite important, should we merge it?

**Comment by softyzhang** [comment]
Yes. Please merge.

File: Detector/DetCRD/compact/CRD_o1_v01/CRD_o1_v01-onlyTracker.xml [hunk 1]
```diff
@@ -32,7 +32,7 @@
   <include ref="../CRD_common_v01/VXD_v01_01.xml"/>
   <include ref="../CRD_common_v01/FTD_SkewRing_v01_01.xml"/>
   <include ref="../CRD_common_v01/SIT_SimplePixel_v01_01.xml"/>
-  <include ref="../CRD_common_v01/DC_Simple_v01_02.xml"/>
+  <include ref="../CRD_common_v01/DC_Simple_v01_03.xml"/>
```
Comment (ihepzhangyao) [inline Detector/DetCRD/compact/CRD_o1_v01/CRD_o1_v01-onlyTracker.xml:35]: I can't find DC_Simple_v01_03.xml. Could you please double check?

File: Detector/DetCRD/compact/CRD_o1_v02/CRD_o1_v02-onlyTracker.xml [hunk 2]
```diff
@@ -31,7 +31,7 @@
   <include ref="../CRD_common_v01/VXD_v01_01.xml"/>
   <include ref="../CRD_common_v01/FTD_SkewRing_v01_01.xml"/>
   <include ref="../CRD_common_v01/SIT_SimplePixel_v01_01.xml"/>
-  <include ref="../CRD_common_v01/DC_Simple_v01_01.xml"/>
+  <include ref="../CRD_common_v01/DC_Simple_v01_03.xml"/>
```
Comment (ihepzhangyao) [inline Detector/DetCRD/compact/CRD_o1_v02/CRD_o1_v02-onlyTracker.xml:34]: DC_Simple_v01_03.xml is missing?

**Comment by ihepzhangyao** [comment]
Could you please post a set of performance check based on this branch?

**Comment by fucd** [comment]
@myliu-hub How about now? I also hope it can be merged. After finish this, can you add extensiton data into DC DetElement, just like TPC?

File: Detector/DetCRD/compact/CRD_o1_v01/CRD_o1_v01-onlyTracker.xml (hunk 1 above)
Comment (mirguest) [reply Detector/DetCRD/compact/CRD_o1_v01/CRD_o1_v01-onlyTracker.xml:35]: Hi @myliu-hub , could you reply Yao's comment?

**Comment by myliu-hub** [comment]
This request has been updated in several aspects:

1. The version of RecGenfitAlgSDT algorithm is updated to the latest
2. The drift chamber of stero wire and straight wire can be constructed using **the same source file**. It only needs to modify the XML file. In the XML file, the skew Angle can be set to **0 degrees** or up to **12 degrees**.
3. Updated track reconstruction algorithm, Silicon detector and drift chamber can be **combined measurement**，`Reconstruction/RecGenfitAlg/src/RecGenfitAlgSDT.cpp`
4. Run the script `Examples/options/tut_detsim_rec_SDT.py` ，you can check the algorithm.

Here is the result of track fitting(single particle): 

1. Momentum resolution：
pt100
2. Spatial resolution(**110um**)：
图片1
3. Mass of higgs(e+e- ->ZH, H -> mu+mu-):

MassH (2)

File: Detector/DetDriftChamber/src/driftchamber/DriftChamber.cpp [hunk 3]
```diff
@@ -1,12 +1,3 @@
-//====================================================================
-//  Detector description implementation of the Drift Chamber
```
Comment (ihepzhangyao) [inline Detector/DetDriftChamber/src/driftchamber/DriftChamber.cpp:2]: Please add the file description.

File: Digitisers/DCHDigi/src/DCHDigiAlg.h [hunk 4]
```diff
@@ -54,38 +56,51 @@ class DCHDigiAlg : public GaudiAlgorithm
   NTuple::Array<int  > m_cell      ;
   NTuple::Array<float> m_cell_x    ;
   NTuple::Array<float> m_cell_y    ;
+  NTuple::Array<float> m_cell1_x   ;
```
Comment (ihepzhangyao) [inline Digitisers/DCHDigi/src/DCHDigiAlg.h:59]: What's the difference between cell and cell1. Could you please add a commit?

**Review by ihepzhangyao — COMMENTED** [review]
Please add some commit to your code.

**Review by ihepzhangyao — COMMENTED** [review]
All review have been cleared

**Review by ihepzhangyao — APPROVED** [review]

**Review by ihepzhangyao — APPROVED** [review]

**Review by mirguest — APPROVED** [review]

**Pull request merged**